test/features/personalization/delayedModal.test.js

```diff
@@ -0,0 +1,74 @@
+import { expect } from '@esm-bundle/chai';
+import { parseUrl, defineDelayedModalParams, decorateDelayedModalAnchor, initDelayedModal } from '../../../libs/features/personalization/personalization.js';
+import { waitForElement } from '../../helpers/waitfor.js';
+
+const DELAYED_MODAL_DISPLAY_MODE = {
+  oncePerPageLoad: 'pageload',
+  oncePerSession: 'session',
+};
+
+it('parses URL properly', () => {
+  expect(parseUrl()).to.deep.equal({});
+  expect(parseUrl('https://www.adobe.com/')).to.deep.equal({
+    hash: '',
+    href: '/',
+    pathname: '/',
+    search: '',
+  });
+  expect(parseUrl('https://www.adobe.com/testpage/?delay=1&display=pageload#delayed-modal')).to.deep.equal({
+    hash: '#delayed-modal',
+    href: '/testpage/?delay=1&display=pageload#delayed-modal',
+    pathname: '/testpage/',
+    search: '?delay=1&display=pageload',
+  });
+});
+
+it('takes the delayed modal parameters from the URL', () => {
+  expect(defineDelayedModalParams()).to.deep.equal({});
+  expect(defineDelayedModalParams('?delay=invalid&display=invalid')).to.be.undefined;
+  expect(defineDelayedModalParams('?delay=1&display=pageload')).to.deep.equal({
+    delay: '1',
+    displayMode: 'pageload',
+    DELAYED_MODAL_DISPLAY_MODE,
+  });
+});
+
+it('add proper attributes and class names to the link', () => {
+  const a = document.createElement('a');
+  decorateDelayedModalAnchor({
+    a,
+    hash: '#delayed-modal',
+    pathname: '/testpage/',
+  });
+  expect(a.getAttribute('href')).to.equal('#delayed-modal');
+  expect(a.getAttribute('data-modal-hash')).to.equal('#delayed-modal');
+  expect(a.getAttribute('data-modal-path')).to.equal('/testpage/');
+  expect(a.getAttribute('style')).to.equal('display: none');
+  expect(a.classList.contains('modal')).to.be.true;
+  expect(a.classList.contains('link-block')).to.be.true;
+  a.remove();
+});
+
+it('creates and opens the delayed modal', async () => {
+  const hash = '#delayed-modal';
```

Comment:
`delayed-modal` shounds to technical.

I wonder if we could adopt a more short/cryptic naming here, like `md` or `dm`.